### ISSUE TITLE
Remove Pandoc dependency

### ISSUE THREAD

**Issue by Peque** [Peque/markdownreveal#6]
Would it make sense?

- https://pypi.python.org/pypi/CommonMark

**Comment by gkmngrgn** [comment]
@Peque I know you don't support Windows 10 yet but docker is bigger dependency than pandoc, and we need Windows 10 Pro version for using Docker over Hyper-v until WSL2 has a stable version.

Can't we use the latest version of pypandoc instead?
https://pypi.org/project/pypandoc/#history

I can help you with this issue but which way is better for you?

**Comment by Peque** [comment]
@gkmngrgn Yeah, Docker is definitely a "bigger" dependency. Only it fixes all other dependency and compatibility issues :smile:

For me, not being able to support Windows, it is an easier approach, but I understand how for Windows users that would be a problem (or maybe not that much, if Docker is supported with WSL2).

If I remember well, the issue with `pypandoc` was that it would not install Pandoc on all platforms. In particular, I think it does not install Pandoc on Linux, I think. If there is a way to avoid this, in example, to be able to `pip install markdownreveal` and have Pandoc installed for all platforms, that would definitely be the best option.

**Comment by gkmngrgn** [comment]
@Peque I think docker is not a too bad idea for us 😋 . @lmelwyn seems already [tried](https://github.com/lmelwyn/markdownreveal) to dockerize your app but I have some feedbacks:
- I think Alpine or any minimal distro could be a better choice than Fedora 26.
- We can prepare an easy command like this one:

```
$ docker run --rm -p 8123:8123 -v $PWD:/ markdownreveal/markdownreveal presentation.md
```

I can create a pull-request including a Dockerfile.

What do you think about using pure markdown implementation written in Python instead? Indeed, pandoc will be installed automatically on MacOS and Windows but as you said, the problem is Linux, and there's no precompiled version of pandoc.

https://github.com/Python-Markdown/markdown

**Comment by Peque** [comment]
@gkmngrgn Yeah, I would gladly accept a PR with a Dockerfile. :blush:

I agree, Fedora 26 is not a great option (specially since it is already discontinued). Alpine seems like a common choice, so I would be fine with that or any other reduced image as long as it is possible and easy to install the required dependencies (Python 3.6+, rsync, Pandoc and npm for Decktape).

**Comment by Peque** [comment]
About substituting Pandoc with another pure-Python package... I would also not mind. But only if it requires few lines of code or low maintenance. I mean, the good thing about Pandoc is not only that it can read Markdown, but also it converts Markdown to reveal.js. Ideally there would be a package for that last step, instead of having to integrate in this project the code for a Markdown-to-reveal.js translator.

**Comment by kiwi0fruit** [comment]
```bash
pip install py-pandoc
```
installs pandoc on Windows x64, macOS and Linux x64. And py-pandoc has the same version as pandoc.
